Panic on discover-baseimage-attestations from gcr.io/distroless

**Describe the bug**
```
podman run --rm -v ./:/code:z  ghcr.io/l3montree-dev/devguard/scanner:main devguard-scanner discover-baseimage-attestations --output /code /code/Containerfile
7:12AM INF commands/discover_baseimage-attestations.go:53 discovering attestations... image=gcr.io/distroless/base-debian13:nonroot@sha256:ced0a2b1936b14d5bddc2ee02a807b1586ca6576a967f5b043f4a3301c8a8f6b
panic: interface conversion: interface {} is string, not map[string]interface {}

goroutine 1 [running]:
github.com/l3montree-dev/devguard/cmd/devguard-scanner/commands.runDiscoverBaseImageAttestations(0x15e7e578fb08, {0x15e7e577b500?, 0x4?, 0x4913f96?})
        github.com/l3montree-dev/devguard/cmd/devguard-scanner/commands/discover_baseimage-attestations.go:77 +0xa9f
github.com/spf13/cobra.(*Command).execute(0x15e7e578fb08, {0x15e7e577b4a0, 0x3, 0x3})
        github.com/spf13/cobra@v1.10.2/command.go:1015 +0xb14
github.com/spf13/cobra.(*Command).ExecuteC(0x8a858a0)
        github.com/spf13/cobra@v1.10.2/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.10.2/command.go:1071
github.com/l3montree-dev/devguard/cmd/devguard-scanner/commands.Execute()
        github.com/l3montree-dev/devguard/cmd/devguard-scanner/commands/root.go:120 +0x1a
main.main()
        github.com/l3montree-dev/devguard/cmd/devguard-scanner/main.go:27 +0x1f
```

**To Reproduce**
Steps to reproduce the behavior:
1. Create a Containerfile with any distroless image: `FROM gcr.io/distroless/base-debian13…`
2. Run `devguard-scanner discover-baseimage-attestations` on the Containerfile
3. Watch panic happen

**Expected behavior**
- If attestations are present and valid (like in the `ghcr.io/l3montree-dev/devguard/scanner:main` itself) a `attestation-bom.json` as output
- In any other case a clear error message what is wrong instead of a panic

--- a/cmd/devguard-scanner/commands/discover_baseimage-attestations.go
+++ b/cmd/devguard-scanner/commands/discover_baseimage-attestations.go
@@ -73,8 +73,10 @@ func runDiscoverBaseImageAttestations(cmd *cobra.Command, args []string) error {
 			predicate = strings.TrimSuffix(predicate, ".json")
 			// DevGuard merges attestations with the same predicate type, so we don't need to include the index in the filename if we have a predicate type
 			attestationFileName = filepath.Join(output, fmt.Sprintf("attestation-%s.json", predicate))
-			// intoto attestation, we can change the content to only include the predicate, since that's the important part
-			attContent = attestation["predicate"].(map[string]any)
+
+			if pred, ok := attestation["predicate"].(map[string]any); ok {
+				attContent = pred
+			}
 		}
 
 		attestationFile, err := os.Create(attestationFileName)

--- a/cmd/devguard-scanner/commands/discover_baseimage-attestations.go
+++ b/cmd/devguard-scanner/commands/discover_baseimage-attestations.go
@@ -33,6 +33,21 @@ func getImageFromContainerFile(containerFile []byte) (string, error) {
 	return imagePath, nil
 }
 
+func attestationOutput(attestation map[string]any, index int) (string, map[string]any) {
+	filename := fmt.Sprintf("attestation-%d.json", index+1)
+	content := attestation
+
+	if predicate, ok := attestation["predicateType"].(string); ok {
+		predicate = strings.Split(predicate, "/")[len(strings.Split(predicate, "/"))-1]
+		predicate = strings.TrimSuffix(predicate, ".json")
+		filename = fmt.Sprintf("attestation-%s.json", predicate)
+		if pred, ok := attestation["predicate"].(map[string]any); ok {
+			content = pred
+		}
+	}
+	return filename, content
+}
+
 func runDiscoverBaseImageAttestations(cmd *cobra.Command, args []string) error {
 	path := args[0]
 	if _, err := os.Stat(path); os.IsNotExist(err) {
@@ -61,23 +76,8 @@ func runDiscoverBaseImageAttestations(cmd *cobra.Command, args []string) error {
 	}
 
 	for i, attestation := range attestations {
-		// try to read the predicate type from the attestation
-		attestationFileName := filepath.Join(output, fmt.Sprintf("attestation-%d.json", i+1))
-
-		attContent := attestation
-
-		if predicate, ok := attestation["predicateType"].(string); ok {
-			// get everything after the last / in the predicate type
-			predicate = strings.Split(predicate, "/")[len(strings.Split(predicate, "/"))-1]
-			// remove .json suffix if it exists
-			predicate = strings.TrimSuffix(predicate, ".json")
-			// DevGuard merges attestations with the same predicate type, so we don't need to include the index in the filename if we have a predicate type
-			attestationFileName = filepath.Join(output, fmt.Sprintf("attestation-%s.json", predicate))
-
-			if pred, ok := attestation["predicate"].(map[string]any); ok {
-				attContent = pred
-			}
-		}
+		filename, attContent := attestationOutput(attestation, i)
+		attestationFileName := filepath.Join(output, filename)
 
 		attestationFile, err := os.Create(attestationFileName)
 		if err != nil {

--- a/cmd/devguard-scanner/commands/discover_baseimage-attestations_test.go
+++ b/cmd/devguard-scanner/commands/discover_baseimage-attestations_test.go
@@ -297,3 +297,62 @@ COPY --from=builder /app /app`,
 		})
 	}
 }
+
+func TestAttestationOutput(t *testing.T) {
+	tests := []struct {
+		name             string
+		attestation      map[string]any
+		index            int
+		expectedFilename string
+		expectedContent  map[string]any
+	}{
+		{
+			name: "intoto attestation: filename from predicateType, only predicate as content",
+			attestation: map[string]any{
+				"_type":         "https://in-toto.io/Statement/v1",
+				"predicateType": "https://cyclonedx.org/bom",
+				"predicate":     map[string]any{"bomFormat": "CycloneDX"},
+			},
+			index:            0,
+			expectedFilename: "attestation-bom.json",
+			expectedContent:  map[string]any{"bomFormat": "CycloneDX"},
+		},
+		{
+			name: "predicate is a string falls back to the whole attestation",
+			attestation: map[string]any{
+				"_type":         "https://in-toto.io/Statement/v0.1",
+				"predicateType": "https://spdx.dev/Document",
+				"predicate":     `{"spdxVersion":"SPDX-2.3"}`,
+			},
+			index:            0,
+			expectedFilename: "attestation-Document.json",
+			expectedContent: map[string]any{
+				"_type":         "https://in-toto.io/Statement/v0.1",
+				"predicateType": "https://spdx.dev/Document",
+				"predicate":     `{"spdxVersion":"SPDX-2.3"}`,
+			},
+		},
+		{
+			name:             "no predicateType falls back to index-based filename and full content",
+			attestation:      map[string]any{"someField": "value"},
+			index:            2,
+			expectedFilename: "attestation-3.json",
+			expectedContent:  map[string]any{"someField": "value"},
+		},
+		{
+			name:             "predicateType present but predicate missing keeps the whole attestation",
+			attestation:      map[string]any{"predicateType": "https://example.com/foo.json"},
+			index:            0,
+			expectedFilename: "attestation-foo.json",
+			expectedContent:  map[string]any{"predicateType": "https://example.com/foo.json"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			filename, content := attestationOutput(tt.attestation, tt.index)
+			assert.Equal(t, tt.expectedFilename, filename)
+			assert.Equal(t, tt.expectedContent, content)
+		})
+	}
+}

